runtime/stdlib/contracts/test.cdc

```diff
@@ -250,4 +258,49 @@ pub contract Test {
         ///
         pub fun useConfiguration(_ configuration: Configuration)
     }
+
+    /// Returns a new matcher that checks if the given test value is either
+    /// a ScriptResult or TransactionResult and the ResultStatus is succeeded.
+    /// Returns false in any other case.
+    ///
+    pub fun beSucceeded(): Matcher {
+        return Matcher(test: fun (value: AnyStruct): Bool {
+            if value.getType() == Type<TransactionResult>() {
+                let result = value as! TransactionResult
+                return result.status == ResultStatus.succeeded
+            } else if value.getType() == Type<ScriptResult>() {
+                let result = value as! ScriptResult
+                return result.status == ResultStatus.succeeded
+            } else {
+                return false
```

Comment:
In `TestContractChecker`, `panic` needs to be declared in the base activation of the checker.

After that, it must be also declared in the base activation of interpreters using the `Test` contract, e.g. in tests in `newTestContractInterpreter`. I think the test runner which uses this test framework, in `github.comcadence-tools/test`, already injects the `panic` function, but we'll need to see.

Reply:
Thanks for the hint :pray: I have added these commit (https://github.com/onflow/cadence/pull/2420/commits/0fea290478be8409ad57597d3690815f46ab754d, https://github.com/onflow/cadence/pull/2420/commits/d1fbaa23bd7d4edddf4084cba9b00755e3533bea).
However, I think we should probably add all these built-in functions (https://github.com/onflow/cadence/blob/master/runtime/stdlib/builtin.go#L36-L52).
I was trying to write some tests for this contract (https://github.com/justjoolz/PRNG/blob/master/cadence/contracts/PRNG.cdc#L78), and it failed when it reached the `getBlock(at: blockHeight)!.id` function call.

Reply:
Yeah, that might be a good idea, maybe as a separate PR.

The language server is doing something like that already, and maybe we can just refactor both at the same time: https://github.com/onflow/cadence-tools/blob/988ec8bbef8b63adb43debf901ec41df8065ff50/languageserver/server/stdlib.go

Basically, we want to use `DefaultStandardLibraryValues`/`DefaultScriptStandardLibraryValues`, and have an implementation of the handler that panics when the functions are used, because they are unavailable.

Reply:
Sounds good :+1:

Reply:
Opened this [issue](https://github.com/onflow/cadence-tools/issues/96), after your comment.